app/client/components/Clipboard.js

```diff
@@ -294,18 +296,54 @@ async function getTextFromClipboardItem(clipboardItem, type) {
 }
 
 /**
- * Helper to determine if the currently active element deserves to keep its own focus, and capture
- * copy-paste events. Besides inputs and textareas, any element can be marked to be a valid
- * copy-paste target by adding 'clipboard_focus' class to it.
+ * Helper to determine if the currently active element deserves to keep its own focus, and capture copy-paste events.
+ *
+ * By default, focus is automatically allowed if:
+ *   - there is no clipboard commands registered,
+ *   - the element is an input, textarea, select or iframe,
+ *   - the element has a tabindex attribute
+ *
+ * You can explicitly allow focus by setting different classes:
+ *   - using the 'clipboard_allow_focus' class will allow focusing the element having the class,
+ *   - using the 'clipboard_allow_group_focus' class will allow focusing any descendant element of the one having the class
+ *
+ * You can explicitly forbid focus by setting the 'clipboard_forbid_focus' class on a element. Forbidding wins over allowing
+ * if both are set.
  */
 function allowFocus(elem) {
-  return elem && (FOCUS_TARGET_TAGS.hasOwnProperty(elem.tagName) ||
+  const {copy, cut, paste} = commands.allCommands;
+  const noCopyPasteCommands = copy._activeFunc === _.noop && cut._activeFunc === _.noop && paste._activeFunc === _.noop;
+  if (elem && elem.classList.contains('clipboard_forbid_focus')) {
+    return false;
+  }
+  if (noCopyPasteCommands) {
+    return true;
+  }
```

Comment:
this is a maybe too optimistic assumption to make (i'd say it's okay since I wrote it but just putting some lights on this), but what this means is: if we notice we don't have any copy/paste commands registered, we assume we don't have to steal usual browser focus. This is the main thing resulting in the default browser focus being possible on non-document pages.